pandas/_libs/lib.pyx

```diff
@@ -2077,11 +2087,18 @@ def maybe_convert_objects(ndarray[object] objects, bint try_float=0,
             else:
                 if not seen.bool_:
                     if seen.datetime_:
-                        if not seen.numeric_:
-                            return datetimes
+                        if not seen.timedelta_:
```

Comment:
same as above

Reply:
@jreback I've proposed simple test in corresponding issue that returns undesired output - should I add this single test only? If not - `maybe_convert_objects` need full test suite (for _all_ possible types combinations) but as I tried to explain earlier  there will be thousands of it which is not doable 'by hands'.

Reply:
we already have quite a few tests
i am ok with systematic testing but don’t go overboard

so happy to merge tests first

Reply:
> so happy to merge tests first

@jreback sorry but from your answer I still can't understand - is single test which is fixed by this PR enough or not?

Reply:
yes this needs at least a single test

if u want to do more comprehensive tests that xfail some (that re not passing now) is ok too (can be before or after this PR)

Reply:
Will add. 

The problem with comprehensive tests:  it cannot be done by hands -> it needs some "trusted" function to tested against i.e. needs copying old function for tests purposes only (as one of the options) - and here I don't understand may I do so or not.